sdk/iothub/azure-iot-deviceprovisioningservice/azure/iot/deviceprovisioningservice/client.py

```diff
@@ -0,0 +1,140 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from typing import Any, Optional, TYPE_CHECKING
+
+from azure.core.pipeline import Pipeline
+from azure.core.pipeline.policies import (
+    BearerTokenCredentialPolicy,
+    ContentDecodePolicy,
+    DistributedTracingPolicy,
+    HeadersPolicy,
+    HttpLoggingPolicy,
+    NetworkTraceLoggingPolicy,
+    RetryPolicy,
+    UserAgentPolicy,
+)
+from azure.core.pipeline.transport import (
+    RequestsTransport,
+)  # pylint: disable=no-name-in-module
+
+from azure.iot.deviceprovisioningservice.auth import SharedKeyCredentialPolicy
+from azure.iot.deviceprovisioningservice._generated._version import VERSION
+from azure.iot.deviceprovisioningservice._generated import (
+    ProvisioningServiceClient as GeneratedProvisioningServiceClient,
+)
+
+from .util.connection_strings import parse_iot_dps_connection_string
+
+if TYPE_CHECKING:
+    # pylint: disable=unused-import,ungrouped-imports
+    from azure.core.credentials import TokenCredential
+
+
+class ProvisioningServiceClient(object):  # pylint: disable=client-accepts-api-version-keyword
+    """
+    API for connecting to, and conducting operations on a Device Provisioning Service instance
+
+    :param str endpoint: The HTTP endpoint of the Device Provisioning Service instance
+    :param credential: The credential type used to authenticate with the Device Provisioning Service instance
```

Comment:
We should also explain why `None` for credential can work i.e. if the endpoint url contains a sas token

Reply:
SAS URL is not supported, http auth header only...so credential is not optional for the service.